File: znld-web/src/main/java/com/sybd/znld/onenet/dto/CommonParams.java
package com.sybd.znld.onenet.dto;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class CommonParams{
    public Integer deviceId;
    public String imei;
    public Integer objId;
    public Integer resId;
    public Integer objInstId;
    public Integer timeout;
    public String toUrlString(){
        return "?imei="+this.imei+"&obj_id="+this.objId+"&res_id="+this.resId+"&obj_inst_id="+this.objInstId+"&timeout="+this.timeout;
    }
}

File: znld-web/src/main/java/com/sybd/znld/onenet/dto/CommonParams.kt
package com.sybd.znld.onenet.dto;

open class CommonParams(var deviceId: Int, var imei: String, var objId: Int, var resId: Int, var objInstId: Int, var timeout: Int) {
    constructor(deviceId: Int, imei: String, key: OneNetKey, timeout: Int): this(deviceId, imei, key.objId, key.resId, key.objInstId, timeout)
    fun toUrlString():String{
        return "?imei=$imei&obj_id=$objId&res_id=$resId&obj_inst_id=$objInstId&timeout=$timeout"
    }
}
